docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+Now let's use this to make a secured request to the /secured/roles-allowed endpoint
+
+.curl Command for /secured/roles-allowed With a token
+[source,shell]
+----
+$ curl -H "Authorization: Bearer 60acf56d-9daf-49ba-b3be-7a423d9c7288" http://127.0.0.1:8080/secured/roles-allowed; echo
+hello + client_id isSecure: false, authScheme: OAuth2
+----
+
+Success! We now have:
+
+* a non-anonymous caller name of client_id
+* an authentication scheme of OAuth2
+
+== Roles mapping
+
+Roles are mapped from one of the claims of the introspection endpoint response. By default, it's the `scope` claim. Roles are obtains by splitting the claim with a space separator. If the claim is an array, no splitting are done, the roles are obtains from the array.
+
+You can customize the name of the claim to use for the roles with the `quarkus.oauth2.role-claim` property.
+
+== Package and run the application
+
+As usual, the application can be packaged using `./mvnw clean package` and executed using the `-runner.jar` file:
+.Runner jar Example
+[source,shell]
+----
+$ ./mvnw clean package
+[INFO] Scanning for projects...
+...
+$ java -jar target/using-oauth2-rbac-runner.jar
+2019-03-28 14:27:48,839 INFO  [io.quarkus] (main) Quarkus 0.20.0 started in 0.796s. Listening on: http://[::]:8080
+2019-03-28 14:27:48,841 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jsonb, security, security-oauth2]
+----
+
+You can also generate the native executable with `./mvnw clean package -Pnative`.
+.Native Executable Example
+[source,shell]
+----
+$ ./mvnw clean package -Pnative
+[INFO] Scanning for projects...
+...
+[using-oauth2-rbac-runner:25602]     universe:     493.17 ms
+[using-oauth2-rbac-runner:25602]      (parse):     660.41 ms
+[using-oauth2-rbac-runner:25602]     (inline):   1,431.10 ms
+[using-oauth2-rbac-runner:25602]    (compile):   7,301.78 ms
+[using-oauth2-rbac-runner:25602]      compile:  10,542.16 ms
+[using-oauth2-rbac-runner:25602]        image:   2,797.62 ms
+[using-oauth2-rbac-runner:25602]        write:     988.24 ms
+[using-oauth2-rbac-runner:25602]      [total]:  43,778.16 ms
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time:  51.500 s
+[INFO] Finished at: 2019-06-28T14:30:56-07:00
+[INFO] ------------------------------------------------------------------------
+
+$ ./target/using-oauth2-rbac-runner
+2019-03-28 14:31:37,315 INFO  [io.quarkus] (main) Quarkus 0.20.0 started in 0.006s. Listening on: http://[::]:8080
+2019-03-28 14:31:37,316 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jsonb, security, security-oauth2]
+----
+
+== Explore the Solution
+
+The solution repository located in the `using-oauth2-rbac` {quickstarts-archive-url}[directory] contains all of the versions we have worked through in this quickstart guide. 
```

Comment:
There's a good chance it won't be true (the fact that you have all the versions?).

Reply:
rephrased